updating html to include badd-row and add-col button forms

vla15 opened the pull request.

speschellk commented on client/js/table-view.js:
```diff
@@ -42,6 +44,18 @@ class TableView {
 		this.renderSumRow();
 	}
 
+	addRow(event) {
```
Set your text editor to use 2-space tabs instead of 4-space tabs and to “convert indentations to spaces”; ask me if you have any trouble. This is key for readability, especially when you checked your code into a central repo that other devs will work off of.

speschellk commented on client/js/test/fixtures/sheet-container.html:
```diff
@@ -1,5 +1,11 @@
+	</form>
 	<table id='sheet-current'>
 		<thead>
 		<tr></tr>
```
Child elements should be 2 spaces indented from their parent element.

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -11,6 +11,20 @@ describe('table view', () => {
 	});
 
+	describe('add row button', () => {
+		it('adds another row to the table body', () => {
```
Make your test assertions describe the full functionality (like cause and effect). If I'm a dev who is less familiar with your code base, and I see that this test is failing, I should know that the button was supposed to "add another row to the table body WHEN CLICKED"

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -11,6 +11,20 @@ describe('table view', () => {
+			const view = new TableView(model);
+			view.init();
+
+			expect(model.numRows).toEqual(3);
```
There is more to test here. Where should the Add Row button add a new row? Top of table? Bottom of table?

speschellk commented on public/js/bundle.js:
```diff
@@ -105,8 +105,10 @@ class TableView {
 		this.headerRowEl = document.querySelector('THEAD TR');
```
Your bundle.js file should not be checked into git or GitHub, since it's a big ugly mess of a file that production environments (Heroku) will generate for you anyway. This is why we put it in the .gitignore file at the beginning of development.

Delete from GitHub and don't commit any modifications to bundle.js in git.

vla15 commented:
Fixed the indentation problem and removed bundle.js from the repository.  I've made some changes to the table view tests as well so they're more descriptive.

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -11,6 +11,50 @@ describe('table view', () => {
 		document.documentElement.innerHTML = html;
 	});
 
+	describe('add col button', () => {
```
This is the only file that didn't update from tabs to spaces.

vla15 commented:
Ah darn it, could've sworn I converted all the files to space indentation.  Pushed the indentation changes to the table-view.test.js file.

speschellk commented on client/js/table-model.js:
```diff
@@ -1,21 +1,21 @@
-		this.numRows = numRows;
-		this.data = {};
-	}
+  constructor(numCols=10, numRows=20) {
```
Please include a space before and after each operator, since that makes your code more human-readable.

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,141 @@ const { getLetterRange, getSum } = require('./array_util');
+  renderTableHeader() {
+    removeChildren(this.headerRowEl);
+    getLetterRange('A', this.model.numCols)
+    .map(colLabel => createTH(colLabel))
```
Indent chained methods 2 spaces in from "parent" methods.

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,141 @@ const { getLetterRange, getSum } = require('./array_util');
+    removeChildren(this.headerRowEl);
+    getLetterRange('A', this.model.numCols)
+    .map(colLabel => createTH(colLabel))
+    .forEach(th => this.headerRowEl.appendChild(th))
```
Missing semicolon

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,141 @@ const { getLetterRange, getSum } = require('./array_util');
+  handleSheetClick(evt) {
+    const col = evt.target.cellIndex;
+    const row = evt.target.parentElement.rowIndex - 1;
+
```
One too many blank lines

speschellk commented on client/js/test/dom-util.test.js:
```diff
@@ -1,57 +1,57 @@
 
-		})
-	});
+    })
```
Missing semicolon

speschellk commented on client/js/test/dom-util.test.js:
```diff
@@ -1,57 +1,57 @@
-	});
+    })
+  });
 })
```
Missing semicolon

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
 
 
-		model.setValue(location, 'foo')
+    model.setValue(location, 'foo')
```
Missing semicolon

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
-		expect(model._getCellId(location)).toEqual('5:3');
-	})
+    expect(model._getCellId(location)).toEqual('5:3');
+  })
```
Missing semicolon

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,139 @@ const { getLetterRange, getSum } = require('./array_util');
+
+  addCol() {
+    event.preventDefault();
+    this.model.numCols++
```
Missing semicolon

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,139 @@ const { getLetterRange, getSum } = require('./array_util');
+          colNumbers.push(parseInt(value, 10));
+        }
+      }
+      colNumbers = colNumbers.filter(num => !isNaN(num))
```
semicolon

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,139 @@ const { getLetterRange, getSum } = require('./array_util');
+      const tr = createTR();
+      for (let col = 0; col < this.model.numCols; col++) {
+        const position = {col: col, row: row};
+        const value = this.model.getValue(position)
```
semicolon

speschellk commented on client/js/table-view.js:
```diff
@@ -2,126 +2,139 @@ const { getLetterRange, getSum } = require('./array_util');
+  handleSheetClick(evt) {
+    const col = evt.target.cellIndex;
+    const row = evt.target.parentElement.rowIndex - 1;
+    this.currentCellLocation = { col: col, row: row }
```
semicolon

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -4,120 +4,164 @@ const fs = require('fs');
+
+      td = trs[4];
+      expect(td.getAttribute('class')).toBe('sum-row');
+
```
Eliminate these blank lines

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -4,120 +4,164 @@ const fs = require('fs');
+      //model.setValue({col: 1, row: 1}, '123');
+      view.init();
+
+      //expect(model.getValue({col: 2, row: 1})).toBe('123');
```
Looks like you can delete these commented-out lines

vla15 commented:
Made further adjustments to the formatting.  Improved on the add-row & add-col button to invoke the actual function rather then just adding a row/col directly to the model

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
 
-		expect(model.getValue(location)).toBe('foo');
+    expect(model.getValue(location)).toBe('foo');
 
```
Remove extra line.

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
 
-		model.setValue(location, 'foo')
+    model.setValue(location, 'foo');
 
```
Remove lines 12 and 13.

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
 
-		expect(model.getValue(location)).toBeUndefined();
+    expect(model.getValue(location)).toBeUndefined();
 
```
Remove one line.

speschellk commented on client/js/test/table-model.test.js:
```diff
@@ -1,18 +1,18 @@
+  it('can set then get a value', () => {
+    const model = new TableModel();
+    const location = { row: 3, col: 5};
 
```
Remove extra line here. For tests, you can keep all the lines up to an "expect" statement together. After an expect statement, if you have more tests to run in the same block, use one space.

speschellk commented on client/js/test/table-view.test.js:
```diff
@@ -4,120 +4,156 @@ const fs = require('fs');
+  beforeEach(() => {
+    const fixturePath = './client/js/test/fixtures/sheet-container.html';
+    const html = fs.readFileSync(fixturePath, 'utf8');
+
```
Remove extra line
